tools/regression_test.py

```diff
@@ -0,0 +1,772 @@
+    """Convert model to onnx and then get metric.
+
+    Args:
+        pipeline_info (dict):  Pipeline info of test yaml.
+        model_cfg_path (Path): Model config file path.
+        checkpoint_path (Path): Checkpoints path.
+        work_dir (Path): A working directory.
+        device_type (str): A string specifying device, defaults to 'cuda'.
+        pytorch_metric (dict): All pytorch metric info.
+        metric_info (dict): Metrics info.
+        report_dict (dict): Report info dict.
+        test_type (sgr): Test type. 'precision' or 'convert'.
+        logger (logging.Logger): Logger.
+        log_path (Path): Path for logger file.
+        backend_file_name (str): backend file save name.
+    """
+    # get backend_test info
+    backend_test = pipeline_info.get('backend_test', False)
+
+    # get convert_image info
+    convert_image_info = pipeline_info.get('convert_image', None)
+    if convert_image_info is not None:
+        input_img_path = \
+            convert_image_info.get('input_img', './tests/data/tiger.jpeg')
+        test_img_path = convert_image_info.get('test_img', None)
+    else:
+        input_img_path = './tests/data/tiger.jpeg'
+        test_img_path = None
+
+    # get sdk_cfg info
+    sdk_config = pipeline_info.get('sdk_config', None)
+    if sdk_config is not None:
+        sdk_config = Path(sdk_config).absolute().resolve()
+
+    if backend_test is False and sdk_config is None:
+        test_type = 'convert'
+
+    metric_name_list = [str(metric) for metric in pytorch_metric]
+    assert len(metric_name_list) > 0
+    metric_all_list = [str(metric) for metric in metric_info]
+    metric_useless = set(metric_all_list) - set(metric_name_list)
+
+    deploy_cfg_path = \
+        Path(pipeline_info.get('deploy_config')).absolute().resolve()
+    backend_name = str(get_backend(str(deploy_cfg_path)).name).lower()
+    infer_type = \
+        'dynamic' if is_dynamic_shape(str(deploy_cfg_path)) else 'static'
+
+    backend_output_path = Path(work_dir). \
+        joinpath(Path(checkpoint_path).parent.parent.name,
+                 Path(checkpoint_path).parent.name,
+                 backend_name,
+                 infer_type,
+                 Path(checkpoint_path).stem)
+    backend_output_path = backend_output_path.absolute().resolve()
+    backend_output_path.mkdir(parents=True, exist_ok=True)
+
+    # convert cmd string
+    cmd_str = f'cd {str(Path().cwd())} && ' \
+              'python3 ./tools/deploy.py ' \
+              f'{str(deploy_cfg_path.absolute().resolve())} ' \
+              f'{str(model_cfg_path.absolute().resolve())} ' \
+              f'{str(checkpoint_path.absolute().resolve())} ' \
+              f'{input_img_path} ' \
+              f'--work-dir {backend_output_path} ' \
+              f'--device {device_type} ' \
+              '--log-level INFO'
+
+    if sdk_config is not None:
+        cmd_str += ' --dump-info'
+
+    if infer_type == 'dynamic' and test_img_path is not None:
```

Comment:
if test_img_path is not None, we pass it. Should ignore infer_type.

Reply:
Done